andrepereiradasilva/joomla-cms
# [Deprecate _QQ_ 3] Mark _QQ_ as deprecated and normalize language ini parsing.

Pull Request for Improvement.

### Summary of Changes

Deprecate `_QQ_` and normalize parsing of ini language files.
- creates a new method JLanguageHelper::parseIniFile and marks LanguagesHelper::parseFile (helper of com_languages) as deprecated.
- makes the necessary adaptations

### Testing Instructions

1. Apply patch

2. Check all language strings show as usual. Including the ones with `"_QQ_"`

3. Confirm language debug still works.
Enable debug language in global config and system debug plugin.manually make some parsing errors in the en-GB.ini file and reload the page

4. Confirm language overrrides still works.
For instance, load a language string that uses escaped quotes (ex: JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED) and make some changes in the override. Save it and load it again to check if ok.

5. Confirm quoting in server side and client side still works (ex: add to isis template index.php)
```php
$app->enqueueMessage('Server side Test 1 (click to proceed to Client side Test 3): ' . JText::_('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED'));
JText::script('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED');
$this->addScriptDeclaration("alert('Client side Test 1: ' + Joomla.JText._('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED'));");
$this->addScriptDeclaration("alert('Client side Test 2: " . JText::_('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED', true) . "');");
$this->addScriptDeclaration("window.addEventListener('click', function () { Joomla.renderMessages({'error': ['Client side Test 3: ' + Joomla.JText._('JLIB_DATABASE_ERROR_DATABASE_UPGRADE_FAILED')]}) });");
```

6. Any other test you could remember

### Documentation Changes Required

None
